lib/mqtt/mqtt.cc

```diff
@@ -954,3 +1053,30 @@ int mqtt_run(Timeout *t, SObj mqttHandle)
 		  return 0;
 	  });
 }
+
+int mqtt_generate_client_id(char *buffer, size_t length)
+{
+	auto characters =
+	  std::string_view(clientIDCharacters, sizeof(clientIDCharacters) - 1);
+
+	if (length > maxClientIDLength || !length)
```

Comment:
Please don't use `!length` to check that an integer is equal to 0.  Also, please put brackets around any sequence of different operators.

```suggestion
	if ((length > maxClientIDLength) || (length == 0))
```